datasource/src/main/scala/quasar/physical/mongo/Mongo.scala

```diff
@@ -119,12 +121,9 @@ class Mongo[F[_]: MonadResourceErr : ConcurrentEffect] private[Mongo](client: Mo
       ))
     ))
 
-    val max = if (maxMemory > MaxBsonBatch) MaxBsonBatch else maxMemory.toLong
-
     getStats map { doc =>
-      val avgObjSize = doc.get("avgObjSize").fold(0)(_.asInt32.getValue)
-      val res = max / avgObjSize.toLong
-      if (res < 1L) 1L else if (res < MaxQueueSize) res else MaxQueueSize
+      val avgObjSize = doc.get("avgObjSize").fold(1)(_.asInt32.getValue)
```

Comment:
Instead of defaulting to a 1-byte average, what do you think about just having a default queue size of, say, 256 elements and using that if we don't know the `avgObjSize`.

I think it's good to be pessimistic here as we don't know anything about the data and in the current case we'll have a queue of `maximumBatchBytes` elements if we are unable to obtain the object size.